packages/cloud-cognitive/src/components/RemoveDeleteModal/RemoveDeleteModal.js

```diff
@@ -1,62 +1,82 @@
-import React, { useState, useRef } from 'react';
-import { Modal, TextInput } from 'carbon-components-react';
+import React, { useState, useRef, forwardRef } from 'react';
+import cx from 'classnames';
+import {
+  ComposedModal,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  TextInput,
+} from 'carbon-components-react';
 import PropTypes from 'prop-types';
 import uuidv4 from '../../global/js/utils/uuidv4';
 
 import { pkg } from '../../settings';
 const componentName = 'RemoveDeleteModal';
 
-export let RemoveDeleteModal = ({
-  body,
-  className,
-  header,
-  iconDescription,
-  inputInvalidText,
-  inputLabelText,
-  inputPlaceholderText,
-  onRequestClose,
-  onRequestSubmit,
-  open,
-  primaryButtonText,
-  resource,
-  secondaryButtonText,
-  subheader,
-  textConfirmation,
-  ...other
-}) => {
-  const [userInput, setUserInput] = useState('');
-  const idRef = useRef(uuidv4());
-  const onChangeHandler = (e) => {
-    setUserInput(e.target.value);
-  };
-  const primaryButtonDisabled = textConfirmation && userInput !== resource;
-  return (
-    <Modal
-      danger
-      className={`${pkg.prefix}-remove-delete-modal ${className}`}
-      iconDescription={iconDescription}
-      open={open}
-      modalHeading={header}
-      modalLabel={subheader}
-      primaryButtonText={primaryButtonText}
-      primaryButtonDisabled={primaryButtonDisabled}
-      secondaryButtonText={secondaryButtonText}
-      onRequestSubmit={onRequestSubmit}
-      onRequestClose={onRequestClose}
-      {...other}>
-      <p className={`${pkg.prefix}-remove-delete-modal-body`}>{body}</p>
-      {textConfirmation && (
-        <TextInput
-          id={`${idRef.current}-confirmation-input`}
-          invalidText={inputInvalidText}
-          labelText={inputLabelText}
-          placeholder={inputPlaceholderText}
-          onChange={onChangeHandler}
+export let RemoveDeleteModal = forwardRef(
+  (
+    {
+      body,
+      className,
+      iconDescription,
+      inputInvalidText,
+      inputLabelText,
+      inputPlaceholderText,
+      label,
+      onRequestClose,
+      onRequestSubmit,
+      open,
+      primaryButtonText,
+      resource,
+      secondaryButtonText,
+      textConfirmation,
+      title,
+      ...rest
+    },
+    ref
+  ) => {
+    const [userInput, setUserInput] = useState('');
+    const idRef = useRef(uuidv4());
+    const onChangeHandler = (e) => {
+      setUserInput(e.target.value);
+    };
+    const primaryButtonDisabled = textConfirmation && userInput !== resource;
+    const blockClass = `${pkg.prefix}--remove-delete-modal`;
+    return (
+      <ComposedModal
+        open={open}
+        ref={ref}
+        danger
+        className={cx(blockClass, className)}
+        {...rest}>
```

Comment:
Just a comment here, I've noticed that most other components add the rest props before any other props. I believe so that the component props will always take precendent

Reply:
you're right 👍